sycl/source/exception.cpp

```diff
@@ -10,9 +10,105 @@
 #include <CL/sycl/context.hpp>
 #include <CL/sycl/exception.hpp>
 
+#include <cstring>
+
 __SYCL_INLINE_NAMESPACE(cl) {
 namespace sycl {
 
+namespace { // anonymous
+char reserved_for_errorcode[] =
+    "0SYCL1234567812345678"; // 17 (string terminator plus 'SYCL' marker plus
+                             // error code)
+std::error_code sycl121_proxy_errorcode = make_error_code(sycl::errc::invalid);
+} // namespace
+
+exception::exception(std::error_code ec, const char *Msg)
+    : exception(static_cast<context *>(nullptr), ec, Msg) {}
+
+exception::exception(std::error_code ec, const std::string &Msg)
+    : exception(static_cast<context *>(nullptr), ec, Msg) {}
+
+// new SYCL2020 constructors
+exception::exception(std::error_code ec)
+    : exception(static_cast<context *>(nullptr), ec, "") {}
+
+exception::exception(int ev, const std::error_category &ecat,
+                     const std::string &what_arg)
+    : exception(static_cast<context *>(nullptr), {ev, ecat}, what_arg) {}
+
+exception::exception(int ev, const std::error_category &ecat,
+                     const char *what_arg)
+    : exception(static_cast<context *>(nullptr), {ev, ecat},
+                std::string(what_arg)) {}
+
+exception::exception(int ev, const std::error_category &ecat)
+    : exception(static_cast<context *>(nullptr), {ev, ecat}, "") {}
+
+exception::exception(context ctx, std::error_code ec,
+                     const std::string &what_arg)
+    : exception(&ctx, ec, what_arg) {}
+
+exception::exception(context ctx, std::error_code ec, const char *what_arg)
```

Comment:
Suggest using Camel case for variables names to align with the code around.

Reply:
I copy pasted these lines from the SYCL2020 spec.  They are the variable names used there.

But I'd be happy to change them to `Ctx`, `EC` and `WhatArg`.  Let me know.

Reply:
Please, change to align with the code around.

Reply:
done